mbed-mesh-api: Add new API for Wi-SUN configuration

artokin opened the pull request:
### Description

Add new API for setting Wi-SUN:

- network name

- regulatory domain, operating mode and operating class

- setting own and trusted certificate
- removing own and trusted certificates

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [x] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers
@mikter , @mikaleppanen 
<!--
    Optional
    Request additional reviewers with @username
-->
### Release Notes

Add new APIs to Wi-SUN network certificate handling and network configuration. This includes:
- network name
- regulatory domain, operating mode and operating class
- setting own and trusted certificate
- removing own and trusted certificates

Commit:
mbed-mesh-api: Add new Wi-SUN API

Add new API for setting Wi-SUN:
 -network name
 -regulatory domain, operating mode and operating class

VeijoPesonen commented on features/nanostack/mbed-mesh-api/mbed-mesh-api/WisunInterface.h:
```diff
@@ -37,6 +37,44 @@ class WisunInterface : public MeshInterfaceNanostack {
+     * \return MESH_ERROR_NONE on success.
+     * \return MESH_ERROR_UNKNOWN in case of failure.
+     * */
+    mesh_error_t network_regulatory_domain_set(uint8_t regulatory_domain = 0xff, uint8_t operating_class = 0xff, uint8_t operating_mode = 0xff);
```
With each of the parameters 0xFF means what, an invalid value?

artokin replied:
yes, a comment added  to explain 0xff.

teetak01 commented on features/nanostack/mbed-mesh-api/mbed-mesh-api/WisunInterface.h:
```diff
@@ -37,6 +37,44 @@ class WisunInterface : public MeshInterfaceNanostack {
+     *
+     * \param network_name Network name as NUL terminated string. Can't exceed 32 characters and can't be NULL.
+     * \return MESH_ERROR_NONE on success.
+     * \return MESH_ERROR_UNKNOWN in case of failure.
```
All fails are Unknown?

VeijoPesonen commented on features/nanostack/mbed-mesh-api/source/wisun_tasklet.c:
```diff
@@ -336,5 +364,54 @@ int8_t wisun_tasklet_network_init(int8_t device_id)
+        int wisun_network_name_len = sizeof(MBED_CONF_MBED_MESH_API_WISUN_NETWORK_NAME);
+        wisun_settings_str.network_name = (char *)ns_dyn_mem_alloc(wisun_network_name_len);
+        if (!wisun_settings_str.network_name) {
+            return -3;
```
Magic number.

mikter commented on features/nanostack/mbed-mesh-api/source/wisun_tasklet.c:
```diff
@@ -336,5 +364,54 @@ int8_t wisun_tasklet_network_init(int8_t device_id)
+
+    strcpy(wisun_settings_str.network_name, network_name_ptr);
+
+    if (wisun_tasklet_data_ptr && wisun_tasklet_data_ptr->tasklet_state == TASKLET_STATE_BOOTSTRAP_READY) {
```
this can be called any time after interface up even during bootstrap. I think in here you should have flexibility for it even before bootstrap is ready

Commit:
mbed-mesh-api: Add new Wi-SUN certificate API

Add new API for setting Wi-SUN:
-Setting own/trusted certificates
-Removing own/trusted certificates

Commit:
mbed-mesh-api: Add API set_file_system_root_path
